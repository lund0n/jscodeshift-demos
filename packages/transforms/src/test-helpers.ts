import { readFile } from 'fs/promises';
import jscodeshift, { Transform } from 'jscodeshift';
import { dirname, resolve } from 'path';
import { fileURLToPath } from 'url';
import { vi } from 'vitest';

export const createTransformer =
  (transformer: Transform) =>
  (source: string, path: string = 'index.js') =>
    transformer(
      { path, source },
      { j: jscodeshift, jscodeshift, report: vi.fn(), stats: vi.fn() },
      {}
    );

export async function readInput(
  filename: string,
  base = resolve(dirname(fileURLToPath(import.meta.url)), '__testfixtures__')
) {
  return await readFile(resolve(base, `${filename}.input.txt`), 'utf-8');
}

export async function readOutput(
  filename: string,
  base = resolve(dirname(fileURLToPath(import.meta.url)), '__testfixtures__')
) {
  return await readFile(resolve(base, `${filename}.output.txt`), 'utf-8');
}
