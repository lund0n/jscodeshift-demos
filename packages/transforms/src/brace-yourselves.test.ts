import { describe, test, expect } from 'vitest';
import dedent from 'dedent';
import { createTransformer, readInput, readOutput } from './test-helpers';
import transformer from './brace-yourselves';

const transform = createTransformer(transformer);

describe('using snapshots', () => {
  test('wraps the "if" in braces', () => {
    const input = dedent`
      if (true) return "Yes"
    `;
    const output = transform(input);
    expect(output).toMatchSnapshot();
  });

  test('ignores wrapping an "if" when already is in block', () => {
    const input = dedent`
      if (true) {
        return "Yes"
      }
    `;
    const output = transform(input);
    expect(output).toMatchSnapshot();
  });
});

describe('comparing strings', () => {
  test('ignores wrapping an "else" when already is in block', () => {
    const input = dedent`
        if (true) {
        return "Yes"
        } else {
        return "No"
        }
    `;

    const output = transform(input);
    expect(output).toEqual(input);
  });
});

describe('using inline snapshots', () => {
  test('wraps the else in braces', () => {
    const input = dedent`
        if (true) {
            return "Yes"
        } else return "No"
  `;

    const output = transform(input);
    expect(output).toMatchInlineSnapshot(`
      "if (true) {
          return \\"Yes\\"
      } else {
          return \\"No\\"
      }"
    `);
  });

  test('ignores wrapping the else when it is an else if', () => {
    const input = dedent`
      if (true) {
        return "Yes"
      } else if (maybe) {
        return "Maybe"
      } else {
        return "No"
      }
    `;

    const output = transform(input);
    expect(output).toEqual(input);
  });
});

describe('using fixtures', () => {
  test('does the proper transform', async () => {
    const input = await readInput('brace-yourselves');
    const expected = await readOutput('brace-yourselves');

    const output = transform(input);

    expect(output).toEqual(expected);
  });
});
