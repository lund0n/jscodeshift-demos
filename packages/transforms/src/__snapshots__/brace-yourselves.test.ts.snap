// Vitest Snapshot v1

exports[`using snapshots > ignores wrapping an "if" when already is in block 1`] = `
"if (true) {
  return \\"Yes\\"
}"
`;

exports[`using snapshots > wraps the "if" in braces 1`] = `
"if (true) {
  return \\"Yes\\"
}"
`;
